view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js

```diff
@@ -502,11 +508,24 @@ define(
                         result.bankLocationId = ko.observable(null);
                     }
 
-                    return result;
-                });
+                    acc.push(result);
+                    return acc;
+                }, []);
 
                 return paymentList;
             },
+            isPaymentMethodSupported: function (paymentMethod) {
+                if (paymentMethod == 'wechatpayWeb') {
+                    return true;
+                }
+                for (var i = 0; i < unsupportedPaymentMethods.length; i++) {
+                    var match =  paymentMethod.match(unsupportedPaymentMethods[i]);
+                    if(match) {
+                        return false;
+                    }
+                }
+                return true;
```

Comment:
We can make this easier to understand by replacing this block with:

```js
return unsupportedPaymentMethods.indexOf(paymentMethod) !== -1;
```

Reply:
Ah, I see. That makes sense, but I'd provide some indication of why this is being done then.

Reply:
@msilvagarcia fair point added some comments.